### ISSUE TITLE
Breaking change :: [ Function] in your popup menu -> Please READ

### ISSUE THREAD

**Issue by ms-jpq** [ms-jpq/coq_nvim#173]
If you see 

![图片](https://user-images.githubusercontent.com/50598611/131422036-0f18b47b-c9fc-4fc2-8f16-ee568e33f0de.png)

---

**Solution**

Either set `let g:coq_settings = { 'display.icons.mode': 'none' }` to disable icons, or [install a supported font](https://www.nerdfonts.com/).

**Comment by ms-jpq** [comment]
I added Icon support, and decided to try my luck turning it on by default.

🐸 

---

If you turn off the icon support, it goes back to same as before, and if you install the fonts:

![icons.png](https://raw.githubusercontent.com/ms-jpq/coq.artifacts/artifacts/preview/icons.png)

**Comment by beauwilliams** [comment]
I am seeing this issue here. We need a space after the icons for when no subtext to it's not juxtaposed next to the ]

![image](https://i.ibb.co/GkZ8vyj/Screen-Shot-2021-09-02-at-10-28-46-am.png)

**Comment by ms-jpq** [comment]
EDITED:

---

##### `coq_settings.display.icons.spacing`

How many ` ` of padding to use after icon.

Increase this if your icons are too close to each other.

**default:**

```json
1
```

**Comment by beauwilliams** [comment]
> @beauwilliams please set `coq_settings.display.pum.kind_context` to `[" [", " ]"]`, or whatever you want it to be.
> 
> I think like for different fonts, the spacing is different? so its probably easier to adjust the brackets than all the icons.
> 
> edit: the default is `[" [", "]"]`, missing the space

Sure thing, thanks for the help and I can add that in no drama.

Should we change the default however?

**Comment by ms-jpq** [comment]
wait i had a brain fart, sorry what i said cannot fix it for you, its in the wrong direction, derp.

ill make a pr to fix it for u

**Comment by ms-jpq** [comment]
https://github.com/ms-jpq/coq_nvim/pull/198

---



##### `coq_settings.display.icons.spacing`

How many ` ` of padding to use after icon.

Increase this if your icons are too close to each other.

**default:**

```json
1
```

**Comment by beauwilliams** [comment]
Good news is the new option works. But it seems I still have the same issue for the BUF type

![error](https://i.ibb.co/cy3KyZn/Screen-Shot-2021-09-02-at-7-49-54-pm.png)

**Comment by beauwilliams** [comment]
If we don't ever have anything but text for the buffer completion type. We could just have it say [aA Text] like the lsp completion at the top there, or add a space when nothing is returned next to the icon

**Comment by ms-jpq** [comment]
sorry please try it now.

**Comment by beauwilliams** [comment]
yep its fixed with latest pull if I set spacing to 2 👍 

![](https://i.ibb.co/HphjjbX/Screen-Shot-2021-09-02-at-8-11-23-pm.png)

**Comment by elkowar** [comment]
is it possible to _only_ show the icon, and hide the text description? the redundancy is kind of unnecessary

**Comment by ms-jpq** [comment]
yes, please see https://github.com/ms-jpq/coq_nvim/blob/coq/docs/DISPLAY.md

**Comment by ms-jpq** [comment]
coq_settings.display.icons.mode -> short

**Comment by matheusAMDS** [comment]
I'm trying the option `let g:coq_settings = { 'display.icons.mode': 'none' }` but it's not working.

**Comment by mkubasz** [comment]
@ms-jpq Short is buggy too. I'm using nerd fonts.

**Comment by mkubasz** [comment]
This is my solution
`  display = { 
    icons = {
      mappings = {
        Text = "",
        Method = "",
        Function = "",
        Constructor = "",
        Field = "ﰠ",
        Variable = "",
        Class = "ﴯ",
        Interface = "",
        Module = "",
        Property = "ﰠ",
        Unit = "塞",
        Value = "",
        Enum = "",
        Keyword = "",
        Snippet = "",
        Color = "",
        File = "",
        Reference = "",
        Folder = "",
        EnumMember = "",
        Constant = "",
        Struct = "פּ",
        Event = "",
        Operator = "",
        TypeParameter = ""
      }
    }
  }`

**Comment by maxwelljens** [comment]
`let g:coq_settings = { 'display.icons.mode': 'none' }` seems to have no effect for me either. I still see icons. I call it before calling `require('coq')`. The plugin otherwise works perfectly.

**Edit:** I gathered from #292 that adding the following to the Lua configuration, before `require('coq')`, works:
```lua
vim.g.coq_settings = {
  display = { icons = { mode = 'none'} }
}
```
I no longer see icons.
